### PR TITLE
Add workflow_dispatch to release workflow

### PR DESCRIPTION
**Pull request by aeisenberg**
This will allow us to create easily accessible artifacts for any branch.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

**Review by adityasharad — APPROVED** [review]
Good idea.
Shall we also remove the `push: branches: '**/actions/**'`? That was a hack I put in to work around the absence of workflow dispatch at the time.

**Comment by aeisenberg** [comment]
Sure. I can remove that.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -6,17 +6,14 @@
...
     # So this workflow is triggered in the following events:
     # - Release event: a SemVer tag, e.g. v1.0.0 or v1.0.0-alpha, is pushed
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+*'
-    # OR
-    # - Test event: this file is modified on a branch in the main repo containing `/actions/` in the name.
-    branches:
-      - '**/actions/**'
     paths:
       - '**/workflows/release.yml'
```
Comment (adityasharad) [inline .github/workflows/release.yml:18]: Sorry, realised this `paths` needs to be removed along with the `branches`, OR even better, we turn it into a `pull_request: paths: '**/workflows/release.yml'`. Then we can test PRs that change the workflow itself without running on every push.

Comment (aeisenberg) [reply .github/workflows/release.yml:18]: Yep. That makes sense.

**Comment by aeisenberg** [comment]
Wait...pull request won't work since this is using a secret from the parent repo. Will running on push work at all? Unless it's running in the parent. Maybe we can use a different event...

**Review by adityasharad — APPROVED** [review]
Much nicer than where we started. Thanks for the patience!

**Comment by aeisenberg** [comment]
`pull_request_target` is what we need, I think, but concerned about privacy implications of exfiltrating secrets. How does that work?

**Comment by adityasharad** [comment]
`pull_request` is fine. I don't expect us to make that secret available and run the release workflow on PRs from forks. If we want to update the release workflow we should open a PR from a branch in the parent repo.

**Comment by aeisenberg** [comment]
This workflow will fail if it is run on `pull_request`, so there's no point in running at all for that event.

**Comment by aeisenberg** [comment]
I'm trying the `pull_request_target` right now. If this works, then

**Pull request merged**